src/boot/bootctl.c

```diff
@@ -2002,11 +2047,14 @@ static int verb_install(int argc, char *argv[], void *userdata) {
 
         (void) sync_everything();
 
-        if (arg_touch_variables)
-                r = install_variables(arg_esp_path,
-                                      part, pstart, psize, uuid,
-                                      "/EFI/systemd/systemd-boot" EFI_MACHINE_TYPE_NAME ".efi",
-                                      install);
+        if (!arg_touch_variables)
+                return r;
+
+        if (arg_arch_all)
+                return log_warning_errno(r, "Not changing EFI variables with --all-architectures.");
```

Comment:
you are propagating `r` here, which has no useful value. You want to return 0 here. 

also, LOG_WARN is too loud of a log level. It's LOG_NOTICE or LOG_INFO at best, since it's nothing actionable at all.

i.e.:

```c
if (arg_arch_all) {
        log_info(…);
        return 0;
}
``